src/string.cr

```diff
@@ -2938,7 +3070,16 @@ class String
     {pre, mid, post}
   end
 
-  def byte_index(byte : Int, offset = 0)
+  # Returns the index of *byte* in the string, or `nil` if the byte is not present.
+  # If *offset* is present, it defines the position to start the search.
```

Comment:
Could you add the note about offset being negative here as well?

Reply:
But this method unlike the others does not support negative offset. It use to_unsafe[index] so it may lead to undefined behavior.. I hadn't noticed it before, we must rewrite it. This PR or another?

Reply:
I think such an small enhancement fits into this PR.

Reply:
@straight-shoota done in last commit.